packages/amplify_storage_s3/android/src/main/kotlin/com/amazonaws/amplify/amplify_storage_s3/types/FlutterDownloadFileRequest.kt

```diff
@@ -47,15 +49,14 @@ data class FlutterDownloadFileRequest(val request: Map<String, *>) {
     }
 
     companion object {
-        fun isValid(request: Map<String, *>): Boolean {
-            var valid = true
-            if (request["path"] !is String) {
-                valid = false
+        private const val validationErrorMessage: String = "DownloadFile request malformed."
+        fun validate(request: Map<String, *>) {
+            if(request["path"] !is String? || request["path"] == null) {
```

Comment:
Isn't the order opposite? You'd first check for null and then unwrap and check for type? On second thought, isn't this is exactly the same as original condition `request["path"] !is String` where null wouldn't match since it's not a `String`?

Reply:
Fixed.